pom.xml

```diff
@@ -307,6 +309,15 @@
                 </plugins>
             </build>
         </profile>
+        <profile>
+            <id>vscode-ide</id>
+            <activation>
+                <property><name>m2e.version</name></property>
```

Comment:
i wonder how is this related to vscode? i fear it could be activated in other cases, too

Reply:
So VSCode and Eclipse both use the same M2M Eclipse plugin and that property is how its detected while the IDE is running.